### ISSUE TITLE
Phase 5: Modelling & SHAP Analysis

### ISSUE THREAD

**Issue by MelvinDY** [MelvinDY/f1-pit-stop-strategy#5]
## Overview
Train three models using scikit-learn Pipelines, evaluate rigorously, and explain predictions with SHAP.

## Skills
Uses `.claude/skills/scikit-learn/SKILL.md`

## Rules
- Always use `Pipeline` objects — no preprocessing outside pipelines
- `random_state=42` everywhere
- Stratified train/test split

### src/model.py
- [ ] `train_evaluate(model, X_train, X_test, y_train, y_test)` → returns dict with accuracy, precision, recall, F1, AUC-ROC
- [ ] `plot_roc_curve(model, X_test, y_test)` → saves to `visuals/`

### Models

**Baseline**
```python
from sklearn.dummy import DummyClassifier
DummyClassifier(strategy='most_frequent', random_state=42)
```

**Logistic Regression**
```python
Pipeline([
    ('preprocessor', ColumnTransformer([...])),
    ('classifier', LogisticRegression(max_iter=1000, random_state=42))
])
```

**XGBoost**
```python
Pipeline([
    ('preprocessor', ColumnTransformer([...])),
    ('classifier', XGBClassifier(random_state=42))
])
# Use early stopping on a validation split
```

### Evaluation
- [ ] Accuracy, Precision, Recall, F1, AUC-ROC for all three models
- [ ] Confusion matrix per model (`sns.heatmap`)
- [ ] ROC curve overlay for all three models → `visuals/roc_curves.png`
- [ ] Results table in notebook:

| Model | Accuracy | AUC-ROC |
|---|---|---|
| Baseline | ? | 0.50 |
| Logistic Regression | ? | ? |
| XGBoost | ? | ? |

### SHAP (XGBoost only)
- [ ] SHAP summary plot → `visuals/shap_summary_plot.png`
- [ ] 2 waterfall plots — one correct prediction, one incorrect prediction

## Acceptance Criteria
- [ ] All three models trained in Pipeline objects
- [ ] `random_state=42` set everywhere
- [ ] Stratified split used
- [ ] Results table in notebook with filled values
- [ ] **XGBoost AUC-ROC ≥ 0.72** (hard requirement from PRD)
- [ ] `visuals/shap_summary_plot.png` saved
- [ ] At least 2 SHAP waterfall plots
- [ ] `visuals/roc_curves.png` saved
- [ ] `src/model.py` contains both required functions

## Reference
See `PRD.md` sections 8.2 and 10, and `TICKET.md` Phase 5.
